src/scheduler/resource_resv.c

```diff
@@ -1346,8 +1351,8 @@ update_resresv_on_end(resource_resv *resresv, char *job_state)
 					next_occr_time = get_occurrence(resresv->resv->rrule,
 						resresv->resv->req_start, resresv->resv->timezone, 2);
 					if (next_occr_time >= 0) {
-						next_occr = find_resource_resv_by_time(resresv->server->resvs,
-							resresv->name, next_occr_time);
+						next_occr = find_resource_resv_by_time_index(resresv->server->resvs,
+							resresv->name, next_occr_time, -1);
```

Comment:
here and other places, why not pass in index of resresv? actually, why not just remove 'name' argument and always search by index?

Reply:
Since standing reservations can occur multiple times in the array, I didn't pass in any index.